system-addon/content-src/asrouter/asrouter-content.jsx

```diff
@@ -68,22 +80,31 @@ class ASRouterUISurface extends React.PureComponent {
   }
 
   componentWillUnmount() {
-    ASRouterUtils.removeMessageListener(this.onMessageFromParent);
+    ASRouterUtils.removeListener(this.onMessageFromParent);
   }
 
   render() {
     const {message} = this.state;
     if (!message.id) { return null; }
-    return (<SimpleSnippet
-      {...message}
-      UISurface={this.props.id}
-      getNextMessage={ASRouterUtils.getNextMessage}
-      onBlock={this.onBlockById(message.id)}
-      sendUserActionTelemetry={ASRouterUtils.sendUserActionTelemetry} />
+    return (<ImpressionsWrapper
+        message={message}
```

Comment:
Does the `ImpressionsWrapper` care about the message? I couldn't see it being used in the component itself

Reply:
It's used by shouldSendImpressionOnUpdate to check whether an update has actually happened (it needs to look at message)